consensus/src/consensus_observer/observer.rs

```diff
@@ -627,27 +629,46 @@ impl ConsensusObserver {
             ordered_proof,
         } = ordered_block.clone();
 
-        // Verify that we have at least one ordered block
-        if blocks.is_empty() {
-            warn!(
+        // Verify the ordered blocks before processing
+        if let Err(error) = self.verify_ordered_blocks(&blocks, &ordered_proof) {
```

Comment:
+1 to what zekun said. if we did `ordered_block.verify(&verifier)` we can possibly avoid the clone above as well.

Reply:
SGTM. Will update 😄